### PR TITLE
[risk=low][RW-7280] Apply the eslint rule @typescript-eslint/no-unused-vars

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/.eslintrc.js [hunk 1]
```diff
@@ -16,13 +16,13 @@ module.exports = {
     ecmaVersion: 6,
     sourceType: 'module',
     ecmaFeatures: {
-      jsx: true
-    }
+      jsx: true,
```
Comment (jmthibault79) [inline ui/.eslintrc.js:19]: looks like my local IntelliJ applied the "prettier" rules here

File: ui/src/testing/react-test-helpers.tsx [hunk 2]
```diff
@@ -73,7 +73,7 @@ export const simulateComponentChange = async (
   value: any
 ) => {
   const primeReactComponent = component as React.Component<
-    { onChange: (PrimeReactChangeEvent) => void },
+    { onChange: (e: PrimeReactChangeEvent) => void },
```
Comment (jmthibault79) [inline ui/src/testing/react-test-helpers.tsx:76]: this was intended to be used

File: ui/src/app/pages/homepage/data-access-requirements.tsx [hunk 3]
```diff
@@ -275,8 +275,8 @@ const getStatusText = (status: AccessModuleStatus) => {
   const { completionEpochMillis, bypassEpochMillis }: AccessModuleStatus =
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/data-access-requirements.tsx:275]: was complaining that these weren't used, so now we use them (as intended)

File: ui/src/app/pages/login/sign-in.spec.tsx [hunk 4]
```diff
@@ -72,6 +66,7 @@ describe('SignInReact', () => {
     createAccountButton.simulate('click');
     wrapper.update();
     const templateImage = wrapper.find('[data-test-id="sign-in-page"]');
```
Comment (jmthibault79) [inline ui/src/app/pages/login/sign-in.spec.tsx:68]: not used - added what II assumed was the intention

File: ui/src/app/routing/app-routing.tsx [hunk 5]
```diff
@@ -279,7 +277,6 @@ export const AppRoutingComponent: React.FunctionComponent<RoutingProps> =
                   <AppRoute
                     path=''
                     exact={false}
-                    intermediaryRoute={true}
```
Comment (jmthibault79) [inline ui/src/app/routing/app-routing.tsx:282]: AppRoute did not actually use this prop.

It's only set in 2 cases, and in both cases the child component does use it.

File: ui/src/app/components/inputs.tsx [hunk 6]
```diff
@@ -408,6 +408,11 @@ export class CheckBox extends React.Component<CheckBoxProps, CheckBoxState> {
       disabled,
       label,
       labelStyle,
+      // RW-7735: determine correct behavior and fix
+      // this isn't used, but removing it changes the contents of otherProps
+      // -> which changes whether the onChange() prop in the input below is overridden
+      // -> which causes it('calls onChange with target checked state') to fail
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (jmthibault79) [inline ui/src/app/components/inputs.tsx:415]: can't enable the rule for this line only

**Review by yonghaoy — APPROVED** [review]
nice!

**Pull request merged**